tests/cwl/workspace-files.feature

```diff
@@ -0,0 +1,27 @@
+# Tests for the presence of the expected workflow files
+
+Feature: Workspace files
+
+    As a researcher,
+    I want to make sure that the workspace produces expected files,
+    so that I can be sure that the workflow outputs are correct.
+
+    Scenario: The workspace contains the expected input files
+        When the workflow is finished
+        Then the workspace should include "code/helloworld.py"
+        And the workspace should include "data/names.txt"
+
+    Scenario: The code used to generate the plot is correct
+        When the workflow is finished
+        Then the file "code/helloworld.py" should contain "message = "Hello " + name + "!\n""
+
+    Scenario: The people are correctly greeted
+        When the workflow is finished
+        Then the workspace should contain "outputs/greetings.txt"
+        And the size of the file "outputs/greetings.txt" should be between 30B and 40B
+        And the file "outputs/greetings.txt" should contain "Hello Jane Doe!"
+        And the file "outputs/greetings.txt" should contain "Hello Joe Bloggs!"
+
+    Scenario: The total workspace size remains within reasonable limits
```

Comment:
The commit log message says:

```
    Separates the test feature files into different subfolders depending on
    the workflow engine (cwl, yadage, snakemake) used by the workflow. This
    is needed to look for log messages that are relevant to the specific
    workflow engine.
```

However, it is only the `run-duration.feature` that is unique, and the other `log-messages.feature` and `workspace-files.feature` are only living in workflow-engine-specific directories.

That is, this PR introduces:

```
tests
├── cwl
│   ├── log-messages.feature
│   └── workspace-files.feature
├── run-duration.feature
├── serial
│   ├── log-messages.feature
│   └── workspace-files.feature
├── snakemake
│   ├── log-messages.feature
│   └── workspace-files.feature
└── yadage
    ├── log-messages.feature
    └── workspace-files.feature
```

I'm not sure it's worth to separate only one test aside?

Two possibilities:

1. Extract more common checks to upper-level directory, as in the Roofit demo, and people will look in two places to find out what tests are defined.
2. Copy-paste the remaining single `run-duration.feature` also to the engine-specific subdirectories, and remove it from top-level, so that people interested in e.g. CWL have only one place to look.

Reply:
Went for option number 2 - I think it looks tidier and more logical, even though at the price of having some duplication.